compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -685,17 +685,14 @@ object RefChecks {
     }
     // Similar to deprecation: check if the symbol is marked with @migration
     // indicating it has changed semantics between versions.
-    if (sym.hasAnnotation(defn.MigrationAnnot) && ctx.settings.Xmigration.value != NoScalaVersion) {
-      val symVersion: scala.util.Try[ScalaVersion] = sym.migrationVersion.get
-      val changed = symVersion match {
-        case scala.util.Success(v) =>
-          ctx.settings.Xmigration.value < v
+    val xMigrationValue = ctx.settings.Xmigration.value
+    if (sym.hasAnnotation(defn.MigrationAnnot) && xMigrationValue != NoScalaVersion) {
+      sym.migrationVersion.get match {
+        case scala.util.Success(symVersion) if xMigrationValue < symVersion =>
+          ctx.warning(SymbolChangedSemanticsInVersion(sym, symVersion, sym.migrationMessage.get))
```

Comment:
You can remove `sym.migrationMessage.get`. It is the same as `symVersion`. I don't know why it was duplicated in the first place

Reply:
I would have to change the final message if that is ok. Currently:
```
${symbol.showLocated} has changed semantics in version $symbolVersion:
$migrationVersion
```
The migrationMessage provides either a version string or an error message from `ScalaVersion.parse`:

```
 def failure = Failure(new NumberFormatException(
      s"There was a problem parsing ${versionString}. " +
       "Versions should be in the form major[.minor[.revision]] " +
       "where each part is a positive number, as in 2.10.1. " +
       "The minor and revision parts are optional."
    ))
```

Reply:
It cannot be a failure since you're in the `Success` case of the match

Reply:
OK